compiler/lib/php_from_rehp.re

```diff
@@ -301,16 +316,29 @@ let rec foldSources = (sourceFolder, curOut, curIn, curRevMappeds, remain) =>
   };
 
 let rec foldStatements =
-        (statementFolder, curOut, curIn, curRevMappeds, remain) =>
+        (
+          statementFolder,
+          origOut: output,
+          curOut: output,
+          curIn: input,
+          curRevMappeds,
+          remain,
+        ) =>
   switch (remain) {
   | [] => (curOut, List.rev(curRevMappeds))
   | [(s, loc), ...tl] =>
-    let (thisOut, thisMapped) = statementFolder(curOut, curIn, s);
+    let (thisOut, thisMapped) =
+      statementFolder(
+        {...curOut, unshielded_continues: origOut.unshielded_continues},
```

Comment:
It wasn't clear when reading this why we needed both `curOut` and `origOut`. This computes the output by folding over a list of statements, and the outputs would track the set of free labels. Why do we need to know the original set of free labels to determine the next set of free labels after examining a list of statements.

Reply:
Can't we just keep appending to the set of free labels?

For example if the `curOutput.unshielded_continues` starts as the empty set `[]`


```
for(;;) {
  continue b;
}
continue c;
```
We first encounter the for loop, which has free labels `[b]`, so we append to `[]`, so the next `curOutput` is `[b]`, then the next statement has free labels `[c]` so we append `[c]` to `[b]` to finally get `[c, b]`.

Reply:
The statement folder shouldn't really even be examining the `curOut.labels` deeply anyways I think. It should only be appending to whatever it is supplied as the curoutput.

Reply:
In this example:

```
for() {
  s1;
  s2;
} 
```

The `out.unshielded_continues` returned by processing s1 should not be considered when processing s2. If it is, then processing this: 

```
for () {
  /* s1 */ if () {
    switch () {
      case 0:
        continue;
    }
  }
  /* s2 */ switch () {
  }
}
```

Will output this:

```
for () {
  /* s1 */ if () {
    continue_label = null
    switch () {
      case 0:
        continue_label = "#";
        break;
    }
    if (continue_label == "#") {
      continue;
    }
  }
  /* s2 */ continue_label = null
  switch () {
  }
  if (continue_label == "#") {
    continue;
  }
}
```

Instead of this, which is actually what it should output:

```
for () {
  /* s1 */ if () {
    continue_label = null
    switch () {
      case 0:
        continue_label = "#";
        break;
    }
    if (continue_label == "#") {
      continue;
    }
  }
  /* s2 */ switch () {
  }
}
```

See the test h1 switch_continue.php. This change was introduced in commit 5fe32e4d20dc0, and you can see that the extra "continue_label" was removed in the output of that function.